src/file.ml

```diff
@@ -46,33 +46,32 @@ module Syntax = struct
     "opamwix-version", OpamPp.ppacc
       (fun c_version t -> { t with c_version}) (fun t -> t.c_version)
       (OpamFormat.V.string -| OpamPp.of_module "version" (module OpamVersion));
-    "ico", OpamPp.ppacc
-      (fun ico t -> { t with c_images = { t.c_images with ico } })
+    "ico", OpamPp.ppacc_opt
+      (fun ico t -> { t with c_images = { t.c_images with ico = Some ico } })
       (fun t -> t.c_images.ico)
-      (OpamFormat.V.string -| OpamPp.of_module "filename" (module OpamFilename));
-    "bng", OpamPp.ppacc
-      (fun bng t -> { t with c_images = { t.c_images with bng } })
-      (fun t -> t.c_images.bng)
-      (OpamFormat.V.string -| OpamPp.of_module "filename" (module OpamFilename));
-    "ban", OpamPp.ppacc
-      (fun ban t -> { t with c_images = { t.c_images with ban } })
+      OpamFormat.V.string;
+    "bng", OpamPp.ppacc_opt
+      (fun dlg t -> { t with c_images = { t.c_images with dlg = Some dlg } })
+      (fun t -> t.c_images.dlg)
+      OpamFormat.V.string;
+    "ban", OpamPp.ppacc_opt
+      (fun ban t -> { t with c_images = { t.c_images with ban = Some ban } })
       (fun t -> t.c_images.ban)
-      (OpamFormat.V.string -| OpamPp.of_module "filename" (module OpamFilename));
+      OpamFormat.V.string;
     "binary-path", OpamPp.ppacc_opt
       (fun bp t -> { t with c_binary_path = Some bp})
       (fun t -> t.c_binary_path)
-      (OpamFormat.V.string -| OpamPp.of_module "filename" (module OpamFilename));
+      OpamFormat.V.string;
     "binary", OpamPp.ppacc_opt
       (fun binary t -> { t with c_binary = Some binary }) (fun t -> t.c_binary)
-      (OpamFormat.V.string);
+      OpamFormat.V.string;
     "embbed-file", OpamPp.ppacc
       (fun file t -> { t with c_embbed_file = file }) (fun t -> t.c_embbed_file)
-      (OpamFormat.V.map_list ~depth:1
-         (OpamFormat.V.string -| OpamPp.of_module "filename" (module OpamFilename)));
+      (OpamFormat.V.map_list ~depth:1 OpamFormat.V.string);
     "embbed-dir", OpamPp.ppacc
       (fun dir t -> { t with c_embbed_dir = dir }) (fun t -> t.c_embbed_dir)
-      (OpamFormat.V.map_list ~depth:1
-         (OpamFormat.V.string -| OpamPp.of_module "dirname" (module OpamFilename.Dir)));
+      (OpamFormat.V.map_list ~depth:2
+        (OpamFormat.V.map_pair OpamFormat.V.string OpamFormat.V.string));
```

Comment:
Why is this a pair ?

Reply:
It is done to specify both: path to directory to embed and what name should it have in installation directory. This is done because sometimes when we need to get things from switch prefix like "<switch-prefix>/share/toto" we want to include it as "share" and not as "toto".